### PR TITLE
NOJIRA: Update the release process docs

### PR DESCRIPTION
**Pull request by gregorydlogan**
Updates the release process documentation, removing a defunct script and making it more clear how we should be creating the changelogs.

### CODE REVIEW

File: docs/guides/developer/docs/release-manager.md
Comment (lkiesow) [inline docs/guides/developer/docs/release-manager.md]: I'm -1 for making release managers responsible for the accuracy of Jira tickets' metadata. For major versions, people would possibily need to go trough hundreds of tickets.

**Review by lkiesow — CHANGES_REQUESTED** [review]

File: docs/guides/developer/docs/release-manager.md
Comment (lkiesow) [reply docs/guides/developer/docs/release-manager.md]: If you really want this to happen, you could automate this and write a script which acts upon merged pull requests and updates the metadata in Jira.

Comment (gregorydlogan) [reply docs/guides/developer/docs/release-manager.md]: It seems to me like there's a lot of pushback in the community for keeping JIRA, so I'm of the opinion that if we're going to keep it we should make the effort to make sure it's *correct* as well.  I'll try and get that script written in time for 5.0  In the meantime let's hold this I guess.

Comment (lkiesow) [reply docs/guides/developer/docs/release-manager.md]: My point is that those who want to keep it should do the work and not those who already spend time on community tasks by volunteering as release managers. If it gets automated that's good. I'm not against correct metadata ;-P

As an alternative to the merge hook, you could also have the script run on the CI machines, automatically executing on builds for release branches.

Comment (gregorydlogan) [reply docs/guides/developer/docs/release-manager.md]: One of those is definitely what we'll do.  When I'll have the time to write the script is a good question though :)

**Comment by lkiesow** [comment]
@gregorydlogan, whatever the final automation will look like, this documentation change will be obsolete and we could close the issue, couldn't we?

Also, note that the current changelog scripts are not based on `fixVersion` anymore but on pull requests. Hence we could, based on that we know that `fixVersion` is mostly broken, also drop the Jira ticket field if we want to. But that's a discussion separate from this pull request.

**Comment by gregorydlogan** [comment]
There's the bit about the Github vs BitBucket artifact upload at the bottom.  I've rolled the fixversion bits out, so this should be good to go now I think?

**Pull request merged**
